src/daft-parquet/src/file.rs

```diff
@@ -223,8 +223,10 @@ impl ParquetReaderBuilder {
         &self.metadata
     }
 
-    pub fn prune_columns<S: ToString + AsRef<str>>(mut self, columns: &[S]) -> super::Result<Self> {
-        self.selected_columns = Some(HashSet::from_iter(columns.iter().map(|s| s.to_string())));
+    pub fn prune_columns<S: AsRef<str>>(mut self, columns: &[S]) -> super::Result<Self> {
+        self.selected_columns = Some(HashSet::from_iter(
+            columns.iter().map(|s| s.as_ref().to_string()),
```

Comment:
this is just calling `to_string` at the end. Should we just keep `to_string`?

Reply:
Decided to just change it to only take slices of Strings and pass around `Vec<String>` everywhere. Let me know how it looks now